Sources/SwiftBSON/JSON.swift

```diff
@@ -1,55 +1,29 @@
+import ExtrasJSON
 import Foundation
 
-/// Enum representing a JSON value, used internally for modeling JSON
-/// during extendedJSON parsing/generation.
-internal enum JSON: Codable {
-    case number(Double)
-    case string(String)
-    case bool(Bool)
-    indirect case array([JSON])
-    indirect case object([String: JSON])
-    case null
-
-    /// Initialize a `JSON` from a decoder.
-    /// Tries to decode into each of the JSON types one by one until one succeeds or
-    /// throws an error indicating that the input is not a valid `JSON` type.
-    internal init(from decoder: Decoder) throws {
-        let container = try decoder.singleValueContainer()
-        if let d = try? container.decode(Double.self) {
-            self = .number(d)
-        } else if let s = try? container.decode(String.self) {
-            self = .string(s)
-        } else if let b = try? container.decode(Bool.self) {
-            self = .bool(b)
-        } else if let a = try? container.decode([JSON].self) {
-            self = .array(a)
-        } else if let d = try? container.decode([String: JSON].self) {
-            self = .object(d)
-        } else if container.decodeNil() {
-            self = .null
-        } else {
-            throw DecodingError.dataCorrupted(
-                DecodingError.Context(
-                    codingPath: decoder.codingPath,
-                    debugDescription: "Not a valid JSON type"
-                ))
-        }
+internal struct JSON {
```

Comment:
I kept this wrapper struct around so that we could add the `ExpressibleByXLiteral` and other conformances to `JSONValue`, which we get from ExtrasJSON.